src/main.cpp

```diff
@@ -32,7 +32,7 @@ static uint celltextcolors[] = {
 };
 #define CUSTOMCOLORIDX 0
 
-enum { TS_VERSION = 20, TS_TEXT = 0, TS_GRID, TS_BOTH, TS_NEITHER };
+enum { TS_VERSION = 21, TS_TEXT = 0, TS_GRID, TS_BOTH, TS_NEITHER, TS_TEXT_LS, TS_GRID_LS, TS_BOTH_LS, TS_NEITHER_LS };

```

Comment:
Rather than all these variants, which create a lot of complexity in code, just define the upper bit of the byte as the selection:

TS_SELECTION_MASK = 0x80

if (byte & TS_SELECTION_MASK) // here we know the cell is selected.

byte &= ~TS_SELECTION_MASK  // clear the bit, so that the rest of the code can remain unchanged.

Reply:
Thanks for your taking your time and for your feedback.
Would the revised commit correspond to your idea?